lib/open_food_network/permissions.rb

```diff
@@ -113,7 +113,9 @@ def visible_subscriptions
     end
 
     def managed_enterprises
-      @managed_enterprises ||= Enterprise.managed_by(@user)
+      return Enterprise.all if admin?
+
+      @user.enterprises
```

Comment:
It's funny, I started thinking "the user knows it's an admin, so why not put the condition in the model?". Then I re-read your commit message and I agree, it should stay here.

But what do you think about wrapping the contents of this method in a block to cache it? (untested)

```suggestion
      @managed_enterprises ||= begin
        next Enterprise.all if admin?

        @user.enterprises
      end
```

Hmm that's kinda ugly. Ok I did a little look and it's not used all that much in the codebase, but it does appear to be called twice by `Permissions::Order::visible_orders_where_values` via:
```
grouping(produced_orders_where_values).
        or(managed_orders_where_values).
```

Loading the orders screen already takes a while. I'm not suggesting this will make a big difference but still I like the idea of caching.. up to you.

Reply:
What would you be caching?

- `admin?` is a simple boolean attribute on the loaded Spree::User or hard-coded `false` on the platform user.
- `Enterprise.all` is a scope. It doesn't issue any queries until the end. It's actually combined in different queries and never executed on its own here. There's no overhead.
- `@user.enterprises` is a scope again that. It's not querying anything.

Am I overlooking something? To me it looks like all these calls are extremely fast, just calling some constructors or even cached associations in ActiveRecord and don't really use any time.